Pride error when including project with subprojects on Windows

We have a standalone project which is a multi project build on its own. When we include this in our pride workspace, then pride fails to initialize the pride workspace settings.gradle. This is caused by the fact that it cannot calculate the relative URI of the subproject based on the windows paths. The File API should be used to translate the path to a proper URI.

Project structure:
nl.planon.assembly
                sdk
                suite

nl.planon.assembly/settings.gradle (include 'sdk', 'suite')

ERROR: Exception:
com.prezi.pride.PrideException: There was a problem during the initialization of the pride. Fix the errors above, and try again with pride init --force
        at com.prezi.pride.cli.PrideInitializer.reinitialize(PrideInitializer.java:118) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.commands.actions.InitAction.initPride(InitAction.java:94) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.commands.actions.InitActionBase.createPride(InitActionBase.java:53) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.commands.InitCommand.executeWithConfiguration(InitCommand.java:86) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.commands.AbstractConfiguredCommand.call(AbstractConfiguredCommand.java:30) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.commands.AbstractConfiguredCommand.call(AbstractConfiguredCommand.java:11) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.PrideCli.execute(PrideCli.java:84) [pride-0.10.jar:na]
        at com.prezi.pride.cli.PrideCli.main(PrideCli.java:27) [pride-0.10.jar:na]
Caused by: java.lang.IllegalArgumentException: Illegal character in opaque part at index 2: F:\HEAD\workspace
        at java.net.URI.create(URI.java:852) ~[na:1.8.0_31]
        at com.prezi.pride.cli.PrideInitializer.writeSettingsForChildren(PrideInitializer.java:143) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.PrideInitializer.createSettingsFile(PrideInitializer.java:136) ~[pride-0.10.jar:na]
        at com.prezi.pride.cli.PrideInitializer.reinitialize(PrideInitializer.java:115) ~[pride-0.10.jar:na]
        ... 7 common frames omitted
Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 2: F:\HEAD\workspace
        at java.net.URI$Parser.fail(URI.java:2848) ~[na:1.8.0_31]
        at java.net.URI$Parser.checkChars(URI.java:3021) ~[na:1.8.0_31]
        at java.net.URI$Parser.parse(URI.java:3058) ~[na:1.8.0_31]
        at java.net.URI.<init>(URI.java:588) ~[na:1.8.0_31]
        at java.net.URI.create(URI.java:850) ~[na:1.8.0_31]
        ... 10 common frames omitted

--- a/pride/src/main/java/com/prezi/pride/cli/PrideInitializer.java
+++ b/pride/src/main/java/com/prezi/pride/cli/PrideInitializer.java
@@ -25,7 +25,6 @@ import org.slf4j.LoggerFactory;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.net.URI;
 import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
@@ -140,7 +139,7 @@ public class PrideInitializer {
 	private void writeSettingsForChildren(File prideRootDir, File settingsFile, String rootProjectName, Set<PrideProjectModel> children) throws IOException {
 		for (PrideProjectModel child : children) {
 			FileUtils.write(settingsFile, "include \'" + rootProjectName + child.getPath() + "\'\n", true);
-			String childProjectRelativePath = URI.create(prideRootDir.getCanonicalPath()).relativize(URI.create(child.getProjectDir())).toString();
+			String childProjectRelativePath = prideRootDir.getCanonicalFile().toURI().relativize(new File(child.getProjectDir()).getCanonicalFile().toURI()).toString();
 			FileUtils.write(settingsFile, "project(\':" + rootProjectName + child.getPath() + "\').projectDir = file(\'" + childProjectRelativePath + "\')\n", true);
 			writeSettingsForChildren(prideRootDir, settingsFile, rootProjectName, child.getChildren());
 		}

--- a/pride/src/main/java/com/prezi/pride/cli/commands/actions/InitActionFromImportedConfig.java
+++ b/pride/src/main/java/com/prezi/pride/cli/commands/actions/InitActionFromImportedConfig.java
@@ -42,11 +42,16 @@ public class InitActionFromImportedConfig extends InitActionBase {
 		if ("-".equals(configLocation)) {
 			importedConfig.load(System.in);
 		} else {
-			URI configUri = URI.create(configLocation);
-			if (configUri.isAbsolute()) {
-				importedConfig.load(configUri.toURL());
+			File configFile = new File(configLocation);
+			if (configFile.isFile()) {
+				importedConfig.load(configFile);
 			} else {
-				importedConfig.load(configUri.getRawPath());
+				URI configUri = URI.create(configLocation);
+				if (configUri.isAbsolute()) {
+					importedConfig.load(configUri.toURL());
+				} else {
+					importedConfig.load(configUri.getRawPath());
+				}
 			}
 		}
 

